src/ObjCRuntime/Class.cs

```diff
@@ -23,14 +23,18 @@ public partial class Class : INativeObject
 #if !COREBUILD
 		public static bool ThrowOnInitFailure = true;
 
-		static Dictionary<Type, IntPtr> type_to_class; // accessed from multiple threads, locking required.
+		static Dictionary<Type, ClassCache> type_to_class; // accessed from multiple threads, locking required.
+		struct ClassCache {
+			internal IntPtr handle;
+			internal bool is_custom_type;
+		}
```

Comment:
I'd rather avoid an extra struct / allocation in that code.

@rolfbjarne can we, like we do in `NSObject` use the LSB of the class handle to store this extra data ?

Reply:
@spouliot yes, we could use the LSB of the class handle to store if it's a custom type or not. I'll have a look at that.